datafusion/core/src/physical_plan/file_format/parquet/row_filter.rs

```diff
@@ -219,7 +219,9 @@ impl<'a> ExprRewriter for FilterCandidateBuilder<'a> {
     fn pre_visit(&mut self, expr: &Expr) -> Result<RewriteRecursion> {
         if let Expr::Column(column) = expr {
             if let Ok(idx) = self.file_schema.index_of(&column.name) {
-                self.required_column_indices.push(idx);
+                if !self.required_column_indices.contains(&idx) {
```

Comment:
🤔  maybe using a Set for the indices would make the intent clearer (and ensure a duplicated column doesn't sneak back in). I don't think it is a huge deal however.